### ISSUE TITLE
Requests show "Build config file is required error" while settings set to "Build only if .travis.yml is present"

### ISSUE THREAD

**Issue by rajsite** [travis-ci/travis-ci#7737]
I recently set-up GitHub Pages Deployment using Travis: https://docs.travis-ci.com/user/deployment/pages/

It is configured to deploy on tagged commits and I am finding that after each tagged commit that results in a deploy that a new rejected Build Request shows up in Travis with the message:

`Build config file is required via repository settings, but config is empty.`

For example the build request right after # 579: https://travis-ci.org/ni/VireoSDK/requests

![allbuilds](https://cloud.githubusercontent.com/assets/1588923/25878924/f6c7b75a-34f3-11e7-9720-e9342496a3ea.PNG)

It can be seen that [the commit](https://github.com/ni/VireoSDK/commit/9b9733f) that triggers the Build Request is the one the deploy script used to push the gh-pages branch.

However, in settings I have `Build only if .travis.yml is present` enabled and indeed there is no .travis.yml present in the gh-pages branch in that commit above. Beyond that my understanding is that builds are not triggered for gh-pages branches unless you opt-in.

![reposettings](https://cloud.githubusercontent.com/assets/1588923/25879095/1eb98044-34f5-11e7-9503-585d69b8e41c.PNG)

My gut tells me this is off and I wonder if the force push the Github Pages script does to gh-pages is interfering but I'll ask anyway: Is it expected behavior that having `Build only if .travis.yml is present` enabled will result in a rejected Build Requests saying `Build config file is required via repository settings, but config is empty.`?

If this is expected behavior, is there a way to make it look less like something happened unexpectedly? Would you recommend an empty .travis.yml file in gh-pages that immediately builds and passes?

**Comment by BanzaiMan** [comment]
This might be a duplicate of https://github.com/travis-ci/travis-ci/issues/7562, which shows a little problem with this deployment provider.

**Comment by rajsite** [comment]
Is it not a more general problem with forced pushed branches? I haven't tested that myself.

**Comment by webknjaz** [comment]
@rajsite
From what I see, this shouldn't be related to Pages provider. Yes, it does a force push to target branch (`gh-pages`).
But it's interaction with GitHub, which results in GitHub triggering a webhook against Travis CI telling it: 'Hey, I've got a new commit in `gh-pages` branch for you'.

After that TravisCI should pull the repo and somehow determine whether you've explicitly opted-in to build `gh-pages` branch.

So how does it determine this _before_ running the build job? I guess it just looks up for `.travis.yml` in `master` branch. You don't have `.travis.yml` stored in `master` and it fails. Mystery solved (hopefully).

Of course it would be logical to use `incoming` branch, which you've set up as a default one for your repo.

@BanzaiMan What do you think about this?

**Comment by mitm001** [comment]
I am having this problem also. Has any progress been made for a fix?

**Comment by webknjaz** [comment]
It's problem with your config @mitm011

**Comment by rajsite** [comment]
This is still an issue, albeit a minor one. It still may be confusing for users to see a bunch of failed requests and not having a way to prevent them.

**Comment by webknjaz** [comment]
@rajsite I explained in my comment above that it's not an issue, but misconfiguration on your side.

**Comment by rajsite** [comment]
I can give the except: gh-pages a shot and keep-history looks useful too, thanks! :)

**Comment by rajsite** [comment]
Hmm no dice, looks like gh-pages builds are still requested and failed by Travis even with the following in the .travis.yml of master:

```
branches:
  except:
    - gh-pages
```

**Comment by rajsite** [comment]
As a side note @webknjaz the gh pages deploy is great! I love being able to see the diff of the generated docs, very cool.

Do you know if there is a way to include either the tag or commit of the build that ran to generate the doc? It's not included in the commit message by default and I didn't see a way to configure that here: https://docs.travis-ci.com/user/deployment/pages/

**Comment by webknjaz** [comment]
No, there's feature requests for changing the message to be clearer. Feel free to open an issue in dpl repo and tag me there describing your use case.

**Comment by webknjaz** [comment]
Regarding requests page, it shows that it's got webhook notification from GitHub, they probably keep it this way for debug reasons.
